src/routes/likernft/book/purchase.ts

```diff
@@ -18,21 +18,26 @@ const router = Router();
 router.get('/:classId/new', async (req, res, next) => {
   try {
     const { classId } = req.params;
-    const { from = '' } = req.query;
+    const { from = '', price = undefined } = req.query;
+    const priceIndex = Number(price) || 0;
```

Comment:
In this case, I would prefer using price_index instead of price, as the latter is more commonly used to refer to the actual price of something
```suggestion
    const { from = '', price_index: priceIndexString = undefined } = req.query;
    const priceIndex = Number(priceIndexString) || 0;
```